test/crash_test.dart

```diff
@@ -50,79 +73,158 @@ void main() async {
+          );
+        });
+      }));
+
+      print('## Found ${packageVersions.length} package versions to scan');
+
+      count = 0;
+      final errors = <String>[];
+      var skipped = 0;
+      await Future.wait(packageVersions.map((pv) async {
+        await pool.withResource(() async {
+          final archiveUrl = Uri.tryParse(pv.archiveUrl);
           if (archiveUrl == null) {
             skipped++;
             return;
           }
           late List<int> archive;
           try {
-            archive = gzip.decode(await c.readBytes(archiveUrl));
+            archive = await c.readBytes(archiveUrl);
           } on http.ClientException {
             skipped++;
             return;
           } on IOException {
             skipped++;
             return;
           }
-          try {
-            await TarReader.forEach(Stream.value(archive), (entry) async {
-              if (entry.name.endsWith('.md')) {
-                late String contents;
-                try {
-                  final bytes = await http.ByteStream(entry.contents).toBytes();
-                  contents = utf8.decode(bytes);
-                } on FormatException {
-                  return; // ignore invalid utf8
-                }
-                try {
-                  markdownToHtml(
-                    contents,
-                    extensionSet: ExtensionSet.gitHubWeb,
-                  );
-                } catch (err, st) {
-                  errors
-                      .add('package:$package/${entry.name}, throws: $err\n$st');
-                }
-              }
-            });
-          } on FormatException {
+
+          final result = await _findMarkdownIssues(
+            pv.package,
+            pv.version,
+            archive,
+          );
+
+          // If tar decoding fails
+          if (result == null) {
             skipped++;
             return;
           }
+
+          errors.addAll(result);
+          result.forEach(print);
         });
         count++;
-        if (DateTime.now().difference(lastStatus) >
-            const Duration(seconds: 30)) {
-          lastStatus = DateTime.now();
-          print('Scanned $count / ${packages.length} (skipped $skipped),'
-              ' found ${errors.length} issues');
-        }
+        status(() =>
+            'Scanned $count / ${packageVersions.length} (skipped $skipped),'
+            ' found ${errors.length} issues');
       }));
 
       await pool.close();
       c.close();
 
+      print('## Finished scanning');
+      print('Scanned ${packageVersions.length} package versions in '
+          '${DateTime.now().difference(started)}');
+
       if (errors.isNotEmpty) {
         print('Found issues:');
         errors.forEach(print);
         fail('Found ${errors.length} cases where markdownToHtml threw!');
       }
     },
-    timeout: const Timeout(Duration(hours: 1)),
+    timeout: const Timeout(Duration(hours: 5)),
```

Comment:
Woof, lol. Quite a task :D

Reply:
Yeah, Github Actions might be slow, and we're just parsing a LOT of markdown files :rofl: